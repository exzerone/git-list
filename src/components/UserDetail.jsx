import React from 'react';
import styled from 'styled-components';

const Overlay = styled.div`
	z-index: 98;
	position: fixed;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	display: flex;
	align-items: center;
	justify-content: center;
	height: 100%;
	width: 100%;
	background-color: rgba(0, 0, 0, 0.3);
`;

const Modal = styled.div`
	z-index: 99;
	position: relative;
	width: 100%;
	max-width: 320px;
	max-height: 100%;
	margin: 0 auto;
`;

const Body = styled.div`
	padding: 20px 24px;
	border-radius: 4px;
	background-color: #e3dbdb;
	border-radius: 25px;
`;

const CloseButton = styled.button`
	cursor: pointer;
	position: absolute;
	right: 20px;
	padding: 5px;
	border: 0;
	height: 30px;
	width: 30px;
	background: none;
	font-size: 15px;
	-webkit-appearance: none;
`;

function UserDetail({ handleClose, ...userData }) {
	return (
		<Overlay className="overlay" onClick={handleClose}>
			<Modal className="modal" onClick={(e) => e.stopPropagation()}>
				<Body className="modal_body">
					<div className="modal_header">
						<img
							href="user image"
							style={{ height: '150px', width: '150px', borderRadius: '25px' }}
							src={userData.avatar_url}
						/>
						<CloseButton onClick={handleClose} className="close_button">X</CloseButton>
					</div>
					<h3>{`Name: ${userData.name}`}</h3>
					<div>{`Username: ${userData.login}`}</div>
					<div>{`Company: ${userData.company}`}</div>
					<div>{`Location: ${userData.location}`}</div>
					<div>{`Type: ${userData.type}`}</div>
					<div>{`Site Admin: ${userData.site_admin}`}</div>
					<div>{`Public Repoistories: ${userData.public_repos}`}</div>
					<div>{`Bio: ${userData.bio}`}</div>
				</Body>
			</Modal>
		</Overlay>
	);
}

export default UserDetail;
