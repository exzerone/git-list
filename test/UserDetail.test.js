import React from 'react';
import { shallow } from 'enzyme';
import UserDetail from '../src/components/UserDetail.jsx';
var data = require('../data.json');

const mockCallBack = jest.fn();
const wrapper = shallow(<UserDetail handleClose={mockCallBack} user={data[0]}/>);

describe('UserDetail', () => {
	it('render without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('render clickable close button', ()=>{
		const button = wrapper.find('.close_button');
		button.simulate('click');
		expect(mockCallBack.mock.calls.length).toEqual(1);
	});
	
});
