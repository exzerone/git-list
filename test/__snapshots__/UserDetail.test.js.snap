// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail render without crashing 1`] = `
<styled.div
  className="overlay"
  onClick={[MockFunction]}
>
  <styled.div
    className="modal"
    onClick={[Function]}
  >
    <styled.div
      className="modal_body"
    >
      <div
        className="modal_header"
      >
        <img
          href="user image"
          style={
            Object {
              "borderRadius": "25px",
              "height": "150px",
              "width": "150px",
            }
          }
        />
        <styled.button
          className="close_button"
          onClick={[MockFunction]}
        >
          X
        </styled.button>
      </div>
      <h3>
        Name: undefined
      </h3>
      <div>
        Username: undefined
      </div>
      <div>
        Company: undefined
      </div>
      <div>
        Location: undefined
      </div>
      <div>
        Type: undefined
      </div>
      <div>
        Site Admin: undefined
      </div>
      <div>
        Public Repoistories: undefined
      </div>
      <div>
        Bio: undefined
      </div>
    </styled.div>
  </styled.div>
</styled.div>
`;
